install/kubernetes/helm/istio/charts/pilot/values.yaml

```diff
@@ -19,6 +19,7 @@ resources:
     memory: 2048Mi
 env:
   PILOT_PUSH_THROTTLE: 100
+  PILOT_ENABLE_PROTOCOL_SNIFFING: false
```

Comment:
Env variables are not intended to be exposed to users. Protocol sniffing may be an exception since it is an alpha feature and we may move it to a proper API. 

However the main problem is that any risky change should be tested and adopted gradually - it must be controlled by a per-pod setting ( meta ), like we're doing with the inbound filterchains.

Reply:
My suggestion: remove this line ( I believe default value is false anyways ). Add a node meta / annotation to indicate that an app is using sniffing - it doesn't have to be turned on globally, users should be able to opt-in or opt-out for individual workloads.

The annotation doc should include a comment that "in case of problems, the feature can be turned off for the entire cluster by setting .... env variable using --set pilot.env.PILOT_SNIFFING=false

( we should also remove the other 2 env variables, to avoid confusing users)

Reply:
Done. Replaced the pilot environment variable with metadata.